src/util.c

```
@@ -1012,23 +1008,15 @@ void ntttcp_bind_to_device(int sockfd, struct ntttcp_stream_client *sc, char if_
 {
         char *log = NULL;
 
-        if (sc == NULL) {
-                ASPRINTF(&log, "invalid arguments to the function %s", __func__);
+        /* When two interfaces share the same subnet, bind() alone does not determine the outgoing interface.
+           The interface selection falls back to the routing table (typically the default route).
+           To enforce interface selection, use SO_BINDTODEVICE to bind to a specific device. */
+        
+        if (setsockopt(sockfd, SOL_SOCKET, SO_BINDTODEVICE, if_name, strlen(if_name)) < 0) {
+                ASPRINTF(&log, "cannot set option SO_BINDTODEVICE for socket[%d] ifname [%s] client_address [%s]", 
+                        sockfd, if_name, sc->client_address);
                 PRINT_INFO_FREE(log);
-                return;
-        }
-
-        if (sc->use_client_address) {
-                /* When two interfaces share the same subnet, bind() alone does not determine the outgoing interface.
-                   The interface selection falls back to the routing table (typically the default route).
-                   To enforce interface selection, use SO_BINDTODEVICE to bind to a specific device. */
-
-                if (setsockopt(sockfd, SOL_SOCKET, SO_BINDTODEVICE, if_name, strlen(if_name)) < 0) {
-                        ASPRINTF(&log, "cannot set option SO_BINDTODEVICE for socket[%d] ifname [%s] client_address [%s]", 
-                                        sockfd, if_name, sc->client_address);
-                        PRINT_INFO_FREE(log);
-                        /* Allow to go through the default interface */
-                }
+                /* Allow to go through the default interface */
```

This is NOT a decision to be made by the function! It is the caller who should decide whether it should go with default interface if this fails or not. 
Also, we -do- want to test to fail if we cannot bind. If I am not mistaken, as it is, we silently use the default interface and that is not the correct behavior.